### PR TITLE
Initial version_checker refactor

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by medariox** [comment]
Fixes https://github.com/pymedusa/Medusa/issues/5726

File: medusa/updater/update_manager.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+from six import text_type
+
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class UpdateManager(object):
+    def __init__(self):
```
Comment (p0psicles) [inline medusa/updater/update_manager.py]: Init is not required here right? It's not doing anything.

File: medusa/updater/update_manager.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    def get_github_repo():
+        return app.GIT_REPO
+
+    @staticmethod
+    def get_update_url():
+        return app.WEB_ROOT + '/home/update/?pid=' + text_type(app.PID)
+
+    def is_latest_version(self):
+        """Compare the current installed version with the remote version."""
+        if LooseVersion(self.newest_version) > LooseVersion(self.current_version):
```
Comment (p0psicles) [inline medusa/updater/update_manager.py]: newest_version and current_version don't refer to anything in this class. Imo it's cleanest to add them in UpdateManager, as methods and raise exception when not implemented subclassed. Like:

```
    def newest_version(self):
        """Docstring"""
        raise NotImplmentedError

    def current_version(self):
        raise NotImplmentedError
```

**Comment by medariox** [comment]
@labrys That's right, it's what we want.

**Comment by labrys** [comment]
@medariox lol I was agreeing with you.  I'm saying its a good design choice.

**Pull request merged**